integrated_channels/integrated_channel/admin/__init__.py

```diff
@@ -0,0 +1,92 @@
+"""
+Django admin integration for integrated_channel app.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django.contrib import admin
+
+from integrated_channels.integrated_channel.models import (
+    EnterpriseCustomerPluginConfiguration, EnterpriseIntegratedChannel
+)
+
+
+@admin.register(EnterpriseIntegratedChannel)
+class EnterpriseIntegratedChannelAdmin(admin.ModelAdmin):
+    """
+    Django admin model for EnterpriseIntegratedChannel.
+    """
+    list_display = (
+        "name",
+        "data_type",
+    )
+
+    list_filter = ("data_type",)
+    search_fields = ("name", "data_type",)
+
+    class Meta(object):
+        model = EnterpriseIntegratedChannel
+
+
+@admin.register(EnterpriseCustomerPluginConfiguration)
+class EnterpriseCustomerPluginConfigurationAdmin(admin.ModelAdmin):
+    """
+    Django admin model for EnterpriseCustomerPluginConfiguration.
+    """
+
+    fields = (
+        "enterprise_customer_uuid",
+        "enterprise_integrated_channel",
+        "active",
+    )
+
+    list_display = (
+        "enterprise_customer_name",
+        "integrated_channel_name",
+        "data_type",
+        "active",
+    )
+
+    readonly_fields = (
+        "integrated_channel_name",
+        "data_type",
+        "enterprise_customer_name",
+    )
+
+    list_filter = ("enterprise_integrated_channel", "active",)
+    search_fields = ("enterprise_customer_name", "integrated_channel_name",)
+
+    class Meta(object):
+        model = EnterpriseCustomerPluginConfiguration
+
+    def integrated_channel_name(self, ent_customer_plugin_config):
+        """
+        Return the name for the attached integrated_channel.
+
+        Args:
+            ent_customer_plugin_config: The instance of EnterpriseCustomerPluginConfiguration
+                being rendered with this admin form.
+        """
+        return ent_customer_plugin_config.enterprise_integrated_channel.name
+
+    def data_type(self, ent_customer_plugin_config):
+        """
+        Return the data_type for the attached integrated_channel.
+
+        Args:
+            ent_customer_plugin_config: The instance of EnterpriseCustomerPluginConfiguration
+                being rendered with this admin form.
+        """
+        return ent_customer_plugin_config.enterprise_integrated_channel.data_type
+
+    def enterprise_customer_name(self, ent_customer_plugin_config):
+        """
+        Return the name for the attached EnterpriseCustomer.
+
+        Args:
+            ent_customer_plugin_config: The instance of EnterpriseCustomerPluginConfiguration
+                being rendered with this admin form.
+        """
+        if ent_customer_plugin_config.enterprise_customer:
+            return ent_customer_plugin_config.enterprise_customer.name
+        else:
+            return ent_customer_plugin_config.enterprise_customer_uuid
```

Comment:
i think it's possible, but not likely. seems simple enough to safeguard against.